### PR TITLE
[Feature][Connector V2] expose configurable options in Iceberg

### PR DESCRIPTION
**Pull request by FWLamb**
## Purpose of this pull request
refer to https://github.com/apache/incubator-seatunnel/issues/3393

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java [hunk 1]
```diff
@@ -20,27 +20,62 @@
...
+        .withDescription(" the iceberg server uri");
+
+    public static final Option<String> KEY_WAREHOUSE = Options.key("warehouse")
+        .stringType()
+        .noDefaultValue()
+        .withDescription(" the iceberg warehouse");
+
+    public static final Option<String> KEY_CASE_SENSITIVE = Options.key("case_sensitive")
+        .stringType()
+        .noDefaultValue()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: `case_sensitive` have a default value 
image

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java [hunk 2]
```diff
@@ -32,12 +34,35 @@
...
+        .withDescription(" the iceberg used snapshot id");
+
+    public static final Option<Long> KEY_USE_SNAPSHOT_TIMESTAMP = Options.key("use_snapshot_timestamp")
+        .longType()
+        .noDefaultValue()
+        .withDescription(" the iceberg used snapshot timestamp");
+
+    public static final Option<Long> KEY_STREAM_SCAN_STRATEGY = Options.key("stream_scan_strategy")
+        .longType()
+        .noDefaultValue()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java]: image

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java [hunk 3]
```diff
@@ -20,27 +20,62 @@
...
-    private static final String KEY_URI = "uri";
-    private static final String KEY_WAREHOUSE = "warehouse";
-    private static final String KEY_CASE_SENSITIVE = "case_sensitive";
+    public static final Option<String> KEY_CATALOG_NAME = Options.key("catalog_name")
+        .stringType()
+        .noDefaultValue()
+        .withDescription(" the iceberg catalog name");
+
+    public static final Option<String> KEY_CATALOG_TYPE = Options.key("catalog_type")
+        .stringType()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: From the document I see this option is an enum type. Please use enum type.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java [hunk 4]
```diff
@@ -32,12 +34,35 @@
...
+        .withDescription(" the iceberg used snapshot id");
+
+    public static final Option<Long> KEY_USE_SNAPSHOT_TIMESTAMP = Options.key("use_snapshot_timestamp")
+        .longType()
+        .noDefaultValue()
+        .withDescription(" the iceberg used snapshot timestamp");
+
+    public static final Option<Long> KEY_STREAM_SCAN_STRATEGY = Options.key("stream_scan_strategy")
+        .longType()
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java]: Enum type?

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java (hunk 1 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: Thank you.I will fix it.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java (hunk 2 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java:64]: Thank you.I will fix it.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java (hunk 3 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: Thank you.I will fix it.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java (hunk 4 above)
Comment (FWLamb) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/SourceConfig.java]: Thank you.I will fix it.

**Comment by EricJoy2048** [comment]
Can you check the options and find which is `bundledRequired` or `exclusive` or `conditional` ? I'm not familiar with Iceberg.

**Comment by FWLamb** [comment]
> Can you check the options and find which is `bundledRequired` or `exclusive` or `conditional` ? I'm not familiar with Iceberg.

Never mind, I will solve this problem.

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSourceFactory.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        return OptionRule.builder()
+            .required(
+                KEY_CATALOG_NAME,
+                KEY_CATALOG_TYPE,
+                KEY_WAREHOUSE,
+                KEY_NAMESPACE,
+                KEY_TABLE
+            )
+            .optional(
+                KEY_URI,
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSourceFactory.java]:
```suggestion
            .conditional(KEY_CATALOG_TYPE, "hive", KEY_URI)
            .optional(
                KEY_FIELDS,
```

Comment (hailin0) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSourceFactory.java]: 1.`KEY_URI ` is required when `KEY_CATALOG_TYPE` is `hive`
2. add `KEY_FIELDS ` to optional

Comment (FWLamb) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/source/IcebergSourceFactory.java]: copy,thx.Is there any parameter bundledRequired?

**Review by hailin0 — DISMISSED** [review]

File: seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java [hunk 6]
```diff
@@ -20,27 +20,62 @@
...
-    private static final String KEY_TABLE = "table";
-    private static final String KEY_URI = "uri";
-    private static final String KEY_WAREHOUSE = "warehouse";
-    private static final String KEY_CASE_SENSITIVE = "case_sensitive";
+    public static final Option<String> KEY_CATALOG_NAME = Options.key("catalog_name")
+        .stringType()
+        .noDefaultValue()
+        .withDescription(" the iceberg catalog name");
+
+    public static final Option<String> KEY_CATALOG_TYPE = Options.key("catalog_type")
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: I think make `catalog_type` as an enum type is better.

Comment (FWLamb) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: @hailin0  It's good to do this. What do you think?

Comment (hailin0) [reply seatunnel-connectors-v2/connector-iceberg/src/main/java/org/apache/seatunnel/connectors/seatunnel/iceberg/config/CommonConfig.java]: > @hailin0 It's good to do this. What do you think?

@FWLamb
agree with you

**Review by EricJoy2048 — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**